comms/core/src/peer_manager/identity_signature.rs

```diff
@@ -136,9 +138,9 @@ impl IdentitySignature {
         features: PeerFeatures,
         addresses: I,
         updated_at: DateTime<Utc>,
-    ) -> CommsChallenge {
+    ) -> DomainSeparatedHasher<CommsChallenge, CommsCorePeerManagerDomain> {
```

Comment:
nit: I think we can return a `[u8;32]` out here - not sure why I returned the `CommsChallenge` type